packages/jaeger-ui/src/utils/guardReducer.tsx

```diff
@@ -22,3 +22,12 @@ export default function guardReducer<TState, TPayload>(fn: (state: TState, value
     return fn(state, payload);
   };
 }
+
+export function guardReducerWithMeta<TState, TPayload, TMeta>(fn: (state: TState, action: { meta: TMeta, payload: TPayload }) => TState) {
+  return function reducer(state: TState, action: Action<TPayload> & { meta: TMeta }) {
+    if (!action.payload || !action.meta) {
+      return state;
+    }
+    return fn(state, action as { meta: TMeta, payload: TPayload });
```

Comment:
IIRC you might be able to avoid the use of `as` here by using a more specific type, than `Action`, from `redux-actions`.

Reply:
the complaint here is that action.payload is `TPayload | undefined` even though the previous if statement should inform typescript that it cannot be `undefined`
We can discuss in our meeting today but it may be a non-issue in the long-term as I'd prefer not to use `redux-actions` and a better async middleware as there are some limitations with the current set up.

Reply:
I will see if I can just provide a better type, probably can